packages/zambdas/src/ehr/update-lab-order-resources/index.ts

```diff
@@ -285,46 +324,105 @@ const handleSpecimenDateChangedEvent = async ({
     throw new Error(`Specimen/${specimenId} not found in ServiceRequest/${serviceRequestId}`);
   }
 
-  const hasSpecimenCollection = specimen.collection;
-  const hasSpecimenDateTime = specimen.collection?.collectedDateTime;
-  const hasSpecimenCollector = specimen.collection?.collector;
-  const specimenCollector = { reference: `Practitioner/${practitionerIdFromCurrentUser}` };
+  const specimenPatchRequest = makeSpecimenPatchRequest(
+    specimen,
+    DateTime.fromISO(date),
+    practitionerIdFromCurrentUser
+  );
 
-  const operations: Operation[] = [];
+  const updateTransactionRequest = await oystehr.fhir.transaction<Specimen>({
+    requests: [specimenPatchRequest],
+  });
 
-  if (hasSpecimenCollection) {
-    operations.push(
-      {
-        op: hasSpecimenCollector ? 'replace' : 'add',
-        path: '/collection/collector',
-        value: specimenCollector,
-      },
-      {
-        op: hasSpecimenDateTime ? 'replace' : 'add',
-        path: '/collection/collectedDateTime',
-        value: date,
-      }
+  return updateTransactionRequest;
+};
+
+// saves sample collection dates
+// saves aoe question entry & marks QR as complete
+// update pre-submission task to complete and create provenance for who did that
+// makes specimen label
```

Comment:
Love the comments. If you want you could make this a docstring instead

Reply:
Like this! The virtue of a docstring is that if someone uses a function elsewhere, and they hover over the name of the function, the IDE will show your docstring. It can be a nice way to explain what a function does and explain what the input are and the outputs. 
```suggestion
/** 
 * saves sample collection dates
 * saves aoe question entry & marks QR as complete
 * update pre-submission task to complete and create provenance for who did that
 *  makes specimen label
 */
```